itest/rust/src/variant_test.rs

```diff
@@ -191,6 +192,32 @@ fn variant_sys_conversion() {
     assert_eq!(v2, v);
 }
 
+#[itest]
+fn varaint_object_nullptr() {
+    // Some Variants that are of type Object actually contain a null pointer.
+    // The should report their type as Nil instead, so that they don't create invalid Gd<T> objects.
+    let variant;
+    unsafe {
+        // Create a variant from a null object pointer.
+        variant = Variant::from_var_sys_init(|self_ptr| {
+            let null_pointer: godot::sys::GDExtensionObjectPtr = std::ptr::null_mut();
+            let converter = godot::sys::builtin_fn!(object_to_variant);
+            converter(
+                self_ptr,
+                std::ptr::addr_of!(null_pointer) as godot::sys::GDExtensionTypePtr,
+            );
+        });
```

Comment:
Could this be reproduced using an actual Godot method? For https://github.com/godot-rust/gdnative/issues/1001 it was `get_node()`, how did you encounter this problem? Would be great if we could get such a `Variant` using one of the `godot::engine` methods, without `unsafe`.

What you're currently doing is writing a test which does the same as the implementation. As such, we're not testing any real-world scenario.